### PR TITLE
chore: remove unused functions and make externally unused functions local

### PR DESCRIPTION
**Pull request by SamMousa**
This cleans up `RosterConfiguration` a bit.

- Remove unused public functions
- Make public functions that are only used via `self:` local instead.

### CODE REVIEW

File: Modules/RosterManager/RosterConfiguration.lua [hunk 1]
```diff
@@ -169,26 +169,20 @@ local TRANSFORMS = {
...
+    local callback = "_validate_" .. option
+    if type(self[callback]) == "function" then
+        local r = self[callback](value)
+        return r
     end
 
-    return result
-end
-
-function RosterConfiguration:Copy(o)
```
Comment (lantisnt) [inline Modules/RosterManager/RosterConfiguration.lua:186]: [Modules/AuctionManager/GUI.lua#L290](https://github.com/ClassicLootManager/ClassicLootManager/blob/master/Modules/AuctionManager/GUI.lua#L290)

Comment (SamMousa) [reply Modules/RosterManager/RosterConfiguration.lua:186]: Restored it.

File: Modules/RosterManager/RosterConfiguration.lua [hunk 2]
```diff
@@ -81,17 +81,13 @@ function RosterConfiguration:New(i)
     -- Dynamic item value
     o._.dynamicValue = false
 
-    -- Additional settings
-    o.hasHardCap = false
```
Comment (lantisnt) [inline Modules/RosterManager/RosterConfiguration.lua:85]: those are used, while not clean, those are  used in Roster.lua

File: Modules/RosterManager/RosterConfiguration.lua [hunk 3]
```diff
@@ -169,18 +161,54 @@ local TRANSFORMS = {
 }
 
 function RosterConfiguration:inflate(data)
-    for i, key in ipairs(self:fields()) do
+    for i, key in ipairs(fields()) do
         self._[key] = TRANSFORMS[key](data[i])
     end
 end
 
-function RosterConfiguration:deflate()
```
Comment (lantisnt) [inline Modules/RosterManager/RosterConfiguration.lua:177]: oh you are going to hate me Samm 

(๑•́ㅿ•̀๑)ᔆᵒʳʳᵞ

`deflate()`
is used in RaidManager LedgerEntries

Comment (SamMousa) [reply Modules/RosterManager/RosterConfiguration.lua:177]: Haha no worries; will get back to it when i can ;-)

**Comment by lantisnt** [comment]
We need to look againif this is needed if we are removing the transforms. With newest changes I dont see any issues there in #417

**Pull request closed without merging**